src/currency/src/server.cpp

```diff
@@ -238,7 +238,16 @@ class CurrencyService final : public oteldemo::CurrencyService::Service
 
 void RunServer(uint16_t port)
 {
-  std::string address("0.0.0.0:" + std::to_string(port));
+  std::string address;
+  const char* ipv6_addr = std::getenv("IPV6_ADDR");
+  
+  if (ipv6_addr != nullptr) {
+    address = std::string(ipv6_addr) + ":" + std::to_string(port);
+    std::cout << "IPv6 ENABLED - binding to " << address << std::endl;
+  } else {
+    address = "0.0.0.0:" + std::to_string(port);
+  }
+
```

Comment:
```suggestion
  ip =  "0.0.0.0";
  
  if (localhost_ip != nullptr) {
    ip = std::string(localhost_ip);
    logger->Info("Overwriting Localhost IP: " + ip);
  }

  address = ip + ":" +  std::to_string(port);

```

This is probably a little simpler and uses the logging library so we'll get and otlp event for it. Note that this specific code assumes the change from `IPV6_ADDR` to `LOCALHOST_IP`